Sources/Crypto/Cipher/CipherAlgorithm.swift

```diff
@@ -40,37 +40,11 @@ public final class CipherAlgorithm {
     /// Only use this if you know what you are doing; use AES-256 GCM otherwise (see https://github.com/vapor/crypto/issues/59).
     public static let aes256cfb: CipherAlgorithm = .init(c: EVP_aes_256_cfb128())
 
-    /// AES-256 GCM cipher. This is the recommended cipher.
-    /// See the global `AES256GCM` constant on usage.
-    public static let aes256gcm: CipherAlgorithm = .init(c: EVP_aes_256_gcm())
```

Comment:
We need to offer deprecated support for this.

Reply:
No, we don't. This code has never actually worked in the first place. So it should never have been referenced anywhere.

Reply:
Do we. It's the same as in the old days, GCM doesn't work without managing the tag. This constant was never tested (and would have never worked, as GCM forces you to do proper AEAD).

Reply:
Even just something like this could be better than removing it outright:

```swift
extension CipherAlgorithm {
    @available(*, unavailable, message: "Use AES256GCM type instead.")
    public static var aes256gcm: CipherAlgorithm { fatalError() }
}
```

Reply:
It's worth noting that this module exposes a number of top-level "globals" like `AES256GCM`. Although the file that defines that constant has changed, the edge user experience has not (assuming you were using the globals). As far as I could tell, those "globals" were the way you access these algorithms.

So `public var AES256GCM: Cipher { return .init(algorithm: .aes256gcm) }` is now in `AuthenticatedCipher` instead. If you peep the tests, you can see that we can just call `AES256GCM.encrypt([...])` as if nothing had changed (except that it now works).

I think a compile time error for anyone who has manually instantiated that class would be better, in line with the thinking MrMage and vzsg have shared.

Reply:
Using `@available(*, unavailable, message:)` does create a compile-time error. Just a tiny bit more descriptive. 

![screen shot 2018-08-09 at 10 36 40 am](https://user-images.githubusercontent.com/1342803/43905953-20d3a0d4-9bc0-11e8-8287-104973a69207.png)

But I tend to agree that it's very unlikely anyone was using that directly.